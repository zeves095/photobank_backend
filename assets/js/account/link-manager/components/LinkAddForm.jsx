import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {submitLink, updateLink} from '../actionCreator';
import FormWrapper from '../../../forms/FormWrapper';
import {getLinkTargets, getChosenResource} from '../selectors';
import {Confirmator} from '../../../common/Confirmator';
import {validateLinkAddForm} from '../../../common/utils/validation';

/**
 * Форма для добавления ссылки
 */
export class LinkAddForm extends React.Component {
  /**
   * Конструктор компонента
   * form_error - Текст ошибки формы
   * defaults - Данные формы по умолчанию
   * confirmatorQuestions - вопросы для компонента Confirmator
   *
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props) {
    super(props);
    this.state={
      form_error: false,
      defaults:{},
      confirmatorQuestions:[
        "Вы уверены?"
      ]
    };
  }

  /**
   * Делает проверку на факт существования выбранного ресурса на момент маунта выбранного ресурса
   */
  componentDidMount(){
    if(this.props.resource_chosen !== null){
      this.handleSetResource();
    }
  }

  /**
   * Делает проверку на факт изменения выбранного ресурса
   */
  componentDidUpdate(prevProps){
    if(prevProps.resource_chosen !== this.props.resource_chosen){
      this.handleSetResource();
    }
  }

  /**
   * Проставляет поле resource для формы из данных redux
   */
  handleSetResource = ()=>{
    let defaults = this.state.defaults;
    let resources = this.props.resource_chosen.map((resource)=>{return resource.id}).join(',');
    defaults.resource = resources;
    this.setState({
      defaults
    });
  }

  /**
   * Обработчик изменений формы. Обновляет состояние формы после изменения поля
   * @param  {Object} data Данные формы
   */
  handleInputChange = (data)=>{
    this.setState({
      defaults:data
    });
  }

  /**
   * Обработчик отправки формы. Отправляет action с данными формы для создания ссылки
   * @param  {Object} data Данные формы
   */
  handleFormSubmit = (data) =>{
    this.props.submitLink(data);
  }

  /**
   * Обработчик ошибок формы.
   * @param  {string[]} errors [description]
   */
  handleFormError = (errors) =>{
    this.setState({
      form_error: errors.length>0
    });
  }

  /**
   * Обработчик потери фокуса формы. Сейчас не испоьзуется, но пусть будет пока.
   * @param  {Object} data Данные формы
   */
  handleFormBlur = (data)=>{
    //this.props.validateLinkAddForm(data);
  }

  /**
   * Создает список вопросов для подтвердждения в компоненте Confirmator. Проверяет, пытается ли юзер создать ссылки для ресурсов, у которых уже есть ссылки, либо добавить ссылки в уже существующую группу
   */
  getConfirmatorQuestions= ()=>{
    let defaults = this.state.defaults;
    let confirmatorQuestions = this.state.confirmatorQuestions;
    if(typeof defaults.target !== "undefined"){
      if(this.props.targets.includes(defaults.target)){
        let existing = this.props.resource_chosen.filter((res)=>{return typeof res.link_exists !== 'undefined' && res.link_exists && res.link_targets.includes(defaults.target)});
        if(existing.length>0){
          confirmatorQuestions[2] = existing.length+" ресурсов уже имеют ссылки. Все равно продолжить?";
        }else{

          confirmatorQuestions.splice(2, 1);
        }
        confirmatorQuestions[1] = "Группа с именем "+defaults.target+" уже существует. Все равно добавить?";
      }else{
        confirmatorQuestions.splice(1, 1);
        confirmatorQuestions.splice(2, 1);
      }
    }
    this.setState({
      confirmatorQuestions
    });
  }

  render() {
    let confirmator = (<Confirmator active={this.props.resource_chosen.length>0} questions={this.state.confirmatorQuestions} prehook={this.getConfirmatorQuestions} onConfirm={()=>{}} inline={false} disabled={this.state.form_error} buttonTitle={"Добавить"} />);
    return (
      <div className={"link-add-form "+(this.props.resource_chosen.length>0?"open":"")}>
        {this.props.form_error !== null?<div className="form-error">{this.props.form_error}</div>:null}
        <FormWrapper form="link-add" onChange={this.handleInputChange} validate={validateLinkAddForm} onBlur={this.handleFormBlur} onSubmit={this.handleFormSubmit} onError={this.handleFormError} defaults={this.state.defaults} submit={confirmator} />
      </div>);
  }

}

const mapStateToProps = (state) =>{
  return {
    link_editing: state.link.link_editing,
    link: state.link.link_editing_id,
    resource_chosen: getChosenResource(state),
    targets: getLinkTargets(state),
    form_error: state.ui.form.link_add.error
  }
}

const mapDispatchToProps = {
  submitLink,
  updateLink,
  //validateLinkAddForm
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkAddForm);
