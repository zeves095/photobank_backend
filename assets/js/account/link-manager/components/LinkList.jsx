import React, {Component} from 'react';
import {connect} from 'react-redux';
import { getLinkTargets } from '../selectors';
import { chooseLink, addLink, fetchLinks, deleteLink, stopEditing } from '../actionCreator'
import {NotificationService} from '../../../services/NotificationService';
import {ModalImage} from '../../../common/ModalImage';
import {Confirmator} from '../../../common/Confirmator';
/**
 * Список существующих ссылок для текущего пользователя
 */
export class LinkList extends React.Component{
  /**
   * Конструктор компонента
   * links - Существующие ссылки для отображения
   * target - Целевая группа ссылок
   * modal_image_url - Ссылка на изображение для отображения в модальном окне предпросмотра
   * confirmatorQuestions - вопросы для компонента Confirmator
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props){
    super(props);
    this.state={
      links:this.props.links,
      target:"Все",
      modal_image_url:"",
      confirmatorQuestions:[
        "Вы уверены?"
      ],
    };
    this.eventHandler = {};
  }

  /**
   * Фильтрует ссылки
   */
  componentDidUpdate(prevProps, prevState){
    if(prevProps.links !== this.props.links || prevState.f !== this.state.f ){
      this.setState({
        links:this.props.links.filter((item)=>{if(!this.state.f) return true; return JSON.stringify(item).includes(this.state.f);})
      });
    }
  }

  /**
   * Обработчик клика на ссылку. Пока не используется, в будущем планируется редактировать выбранную ссылку по клику
   * @param  {Number} id Идентификатор ссылки
   */
  handleLinkClick = (id)=>{
    //this.props.chooseLink(id);
  }

  /**
   * Обработчик добавления новой ссылки
   * @param  {Event} e Событие клика
   */
  handleLinkAdd = (e)=>{
    this.props.editing||this.props.adding?this.props.stopEditing():this.props.addLink();
  }

  /**
   * Обработчик клика но группу ссылок
   * @param  {Event} e Событие клика
   */
  handleTargetChoice = (e)=>{
    e.preventDefault();
    this.setState({
      target:e.target.dataset['target']
    });
  }

  /**
   * Обработчик удаления ссылки
   * @param  {Number} id Идентификатор ссылки для удаления
   */
  handleLinkDelete = (id)=>{
    this.props.deleteLink(id);
  }

  /**
   * Обработчик копирования всех отображаемых ссылок в буфер обмена
   */
  handleCopyAllToClipboard = ()=>{
    let links = [];
    this.state.links.forEach((link)=>{
      if(link.target !== this.state.target && this.state.target !== "Все"){return false;}
      links.push('https://photobank.domfarfora.ru'+link.external_url);
    });
    if(typeof navigator.clipboard !== "undefined"){navigator.clipboard.writeText(links.join(",\n"))}else{NotificationService.throw('clipboard-error')};
    NotificationService.toast("link-copied");
  }

  /**
   * Обработчик скачивания txt-файла с текстом всех отображаемых ссылок
   */
  handleGetTxt =()=>{
    let links = this.state.links
    .filter((link)=>{
        return link.target == this.state.target || this.state.target == "Все";
    })
    .map((link)=>{
        return(link.link_id);
      }
    );
    if(links.length === 0){NotificationService.toast("custom", "Нет ссылок"); return;}
    let linkTxtForm = document.createElement("form");
    linkTxtForm.target = "_blank";
    linkTxtForm.method = "POST";
    linkTxtForm.action = "/api/links/txt/";
    let linkTxtInput = document.createElement("input");
    linkTxtInput.type = "text";
    linkTxtInput.name = "links";
    linkTxtInput.value = links;
    linkTxtForm.appendChild(linkTxtInput);
    document.body.appendChild(linkTxtForm);
    linkTxtForm.submit();
    linkTxtForm.remove();
  }

  /**
   * Обработчик открытия модального окна с превью изображения ресурса
   * @param  {Number} link Идентификатор ссылки
   */
  handleModalImage = (link)=>{
    this.setState({
      modal_image_url: link
    });
  }

  /**
   * Обработчик закрытия модального окна
   */
  handleModalClose = ()=>{
    this.setState({
      modal_image_url: ""
    });
  }

  /**
   * Подписчик на события на странице для отлавливания комбинаций клавиш
   * @param  {Event} e Событие
   */
  listener = (e)=>{
    e = e || window.event;
    var key = e.which || e.keyCode; // keyCode detection
    var ctrl = e.ctrlKey ? e.ctrlKey : ((key === 17) ? true : false); // ctrl detection

    if ( key == 67 && ctrl ) {
        this.handleCopyAllToClipboard();
      }else if ( key == 83 && ctrl ) {
        e.preventDefault();
        this.handleGetTxt();
        return false;
      }
  }

  /**
   * Обработчик появления курсора мыши на области компонента, для ограничения событий нажатия клавиатурных сокращений
   */
  handleFocus = ()=>{
    document.body.addEventListener("keydown",this.listener);
  }

  /**
   * Обработчик исчезновения курсора мыши с области компонента, для ограничения событий нажатия клавиатурных сокращений
   */
  handleBlur = ()=>{
    document.body.removeEventListener('keydown', this.listener);
  }

  render(){
    let filter = <input type="search" className="link-filter" onChange={(e)=>{this.setState({f : e.target.value})}}  placeholder="Фильтр"/>
    let links = this.state.links.map(
      (link)=>{
        if(link.target !== this.state.target && this.state.target !== "Все"){return false;}
        let delete_btn = <i className="fas fa-trash-alt"></i>;
        let thumb = this.props.thumbs.find((thumb)=>thumb.id === link.resource_id);
        return(
          <div data-linkid={link.link_id} key={"link"+link.link_id} className="link " onClick={()=>{this.handleLinkClick(link.link_id)}}>
            <Confirmator questions={this.state.confirmatorQuestions} prehook={()=>{}} active={true} onConfirm={()=>{this.handleLinkDelete(link.link_id)}} inline={true} customClass={"delete-link"} disabled={false} buttonTitle={delete_btn} />
            <div className="link-info">
            <div><b>Ссылка:</b><a href={'https://photobank.domfarfora.ru'+link.external_url} target="_blank" >{'https://photobank.domfarfora.ru'+link.external_url}</a></div>
          <div><b>Товар: </b>{link.item_name}({link.item_id})</div>
          </div>
        <span className={"resource-preview"+(typeof thumb === 'undefined'?" resource-preview--loading":"")} style={{backgroundImage:typeof thumb === 'undefined'?"none":"url(/catalogue/node/item/resource/"+thumb.thumb_id+".jpg)"}} onClick={(e)=>{e.stopPropagation();this.handleModalImage("/catalogue/node/item/resource/"+thumb.id+".jpg")}}></span>
          </div>
        )
      }
    );
    let tabs = ["Все"].concat(this.props.targets).map((target)=>{
        return(
          <button key={"btn-"+target} className={(target===this.state.target?" active":"")} data-target={target} onClick={this.handleTargetChoice}>{target}</button>
        )
    })
    return(
      <div className={"link-list "+(this.props.adding||this.props.editing?"shrunk":"")} onMouseEnter={this.handleFocus} onMouseLeave={this.handleBlur}>
        {this.state.modal_image_url !== ""?<ModalImage image_url={this.state.modal_image_url} closeModalHandler={this.handleModalClose} />:null}
        <div className="component-header">
          <h2 className="component-title">
            Ссылки
          </h2>
          {filter}
        </div>
        <div className={"component-body" + (this.props.loading?" loading":"")}>
          <div className="component-body__top-section">
            <div className="button-block">
              <button key="btn-add" onClick={this.handleLinkAdd} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button">{this.props.adding||this.props.editing?(<span><i className="fas fa-ban"></i>Отмена</span>):(<span><i className="fas fa-plus-circle"></i>Добавить</span>)}</button>
            <button key="btn-copy" onClick={this.handleCopyAllToClipboard} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button"><i className="fas fa-copy"></i>Скопировать в буфер {!this.props.adding&&!this.props.editing&&(<span className="help-text">&lt;Ctrl&gt;+&lt;C&gt;</span>)}</button>
          <button key="btn-dl" onClick={this.handleGetTxt} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button"><i className="fas fa-align-justify"></i>Скачать ссылки {!this.props.adding&&!this.props.editing&&(<span className="help-text">&lt;Ctrl&gt;+&lt;S&gt;</span>)}</button>
            </div>
            <div className="link-list__tabs button-block">
              {tabs}
            </div>
            {links.length==0?(<div className="resource plaque warning"><i className="fas fa-times-circle"></i>Нет ссылок</div>):links}
          </div>
        </div>
      </div>
    );
  }

}

const mapStateToProps = (state) =>{
  return {
    links: state.link.links_done,
    editing: state.link.link_editing,
    adding: state.link.link_adding,
    targets: getLinkTargets(state),
    thumbs: state.resource.resources_thumbnails,
    loading: state.ui.loading.link_list
  }
}

const mapDispatchToProps = {
    chooseLink,
    addLink,
    fetchLinks,
    deleteLink,
    stopEditing
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkList);
