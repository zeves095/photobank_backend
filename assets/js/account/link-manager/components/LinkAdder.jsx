import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import LinkAddForm from './LinkAddForm';
import LinkResource from './LinkResource';
import ResourceExplorer from './ResourceExplorer';

import { stopEditing } from '../actionCreator'

/**
 * Правые две колонки интерфейса добавления ссылок, включает в себя поиск ресурсов и форму добавления ссылки
 */
export class LinkAdder extends React.Component{

  /**
   * Конструктор компонента
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props){
    super(props);
    this.state={};
  }

  /**
   * Обработчик скрытия этой части интерфейса
   */
  handleCloseAdder = ()=>{
    this.props.stopEditing();
  }

  render(){
    let link_adder = (
      <div className={"link-adder__inner "+(this.props.link_adding?"open":"")}>
        <div className="component-header component-header--subcomponent adder-controls">
          <div className="button-block"><button type="button" name="button" onClick={this.handleCloseAdder}><i className="fas fa-times-circle"></i></button></div>
        </div>
      <div className="component-body__bottom-section resource-section">
        <ResourceExplorer />
      </div>
      <div className="component-body__top-section link-section">
        <h2 className="component-title">
          Добавление ссылки
        </h2>
        <LinkAddForm />
        <LinkResource />
      </div>
    </div>
    )
    return (
      <div className={"link-adder"+(this.props.link_adding?" open":"")}>
        <div className="component-body">
          {this.props.link_adding?link_adder:null}
        </div>
      </div>
    );
  }

}

const mapStateToProps = (state) =>{
  return {
    link_adding: state.link.link_adding,
  }
}

const mapDispatchToProps = {
  stopEditing
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkAdder);
