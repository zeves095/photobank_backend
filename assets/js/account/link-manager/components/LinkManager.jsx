import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {init} from '../actionCreator'
import LinkList from './LinkList';
import LinkAdder from './LinkAdder';

/**
 * Приложение для менеджмента ссылок для внешнего доступа
 */
export class LinkManager extends React.Component{
  /**
   * Конструктор компонента
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props){
    super(props);
    this.state={};
  }

  /**
   * Запрашивает доступные типы файлов, пресетов и информацию о текущем пользователе с сервера
   * @return {[type]} [description]
   */
  componentDidMount(){
    this.props.init();
  }

  render(){
    return(
      <div className="link-manager">
        <div id="notification-overlay">

        </div>
        <div className="flex-wrapper">
            <LinkList />
            <LinkAdder />
        </div>
      </div>
    )
  }

}

const mapStateToProps = (state) =>{
  return {
  }
}

const mapDispatchToProps = {
  init
}

export default connect(mapStateToProps,mapDispatchToProps)(LinkManager);
