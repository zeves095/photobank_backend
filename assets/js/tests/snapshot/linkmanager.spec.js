import React from 'react';
import { LinkManager } from '../../account/link-manager/components/LinkManager';
import { LinkAdder } from '../../account/link-manager/components/LinkAdder';
import { LinkList } from '../../account/link-manager/components/LinkList';
import { LinkAddForm } from '../../account/link-manager/components/LinkAddForm';
import { LinkResource } from '../../account/link-manager/components/LinkResource';
import { ResourceExplorer } from '../../account/link-manager/components/ResourceExplorer';
import { ResourceSearchForm } from '../../account/link-manager/components/ResourceSearchForm';
import { ResourceSearchResults } from '../../account/link-manager/components/ResourceSearchResults';
import Enzyme, {render, shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {store} from '../../account/link-manager/store/';
import {Provider} from 'react-redux';

Enzyme.configure({adapter: new Adapter()});

describe('snapshot testing', ()=>{
    it('LinkManager рендерится правильно', () => {
      const tree = render(<Provider store={store}><LinkManager /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('LinkAdder рендерится правильно', () => {
      const tree = render(<Provider store={store}><LinkAdder /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('LinkList рендерится правильно', () => {
      const tree = render(<Provider store={store}><LinkList links={[]} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('LinkAddForm рендерится правильно', () => {
      const tree = render(<Provider store={store}><LinkAddForm resource_chosen={[]} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('LinkResource рендерится правильно', () => {
      const tree = render(<Provider store={store}><LinkResource resources={[]} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('ResourceExplorer рендерится правильно', () => {
      const tree = render(<Provider store={store}><ResourceExplorer /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('ResourceSearchForm рендерится правильно', () => {
      const tree = render(<Provider store={store}><ResourceSearchForm /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('ResourceSearchResults рендерится правильно', () => {
      const tree = render(<Provider store={store}><ResourceSearchResults resources_found={[]} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
  }
)
