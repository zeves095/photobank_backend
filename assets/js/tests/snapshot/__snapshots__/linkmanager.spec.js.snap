// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing LinkAddForm рендерится правильно 1`] = `"<div class=\\"form-error\\"></div><div class=\\"form-wrapper\\"><form class=\\"rjsf\\"><div class=\\"form-group field field-object\\"><fieldset id=\\"root\\"><div class=\\"form-group field field-object\\"><div class=\\"size-field\\"><div class=\\"button-block\\"><button type=\\"button\\">&#x423;&#x43A;&#x430;&#x437;&#x430;&#x442;&#x44C; &#x440;&#x430;&#x437;&#x43C;&#x435;&#x440; &#x438;&#x437;&#x43E;&#x431;&#x440;&#x430;&#x436;&#x435;&#x43D;&#x438;&#x44F;</button></div></div><div></div><div></div></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_target\\">&#x413;&#x440;&#x443;&#x43F;&#x43F;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_target\\" label=\\"&#x413;&#x440;&#x443;&#x43F;&#x43F;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_max_requests\\" value></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_expires_by\\" value></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_access\\" value></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource\\" value></div></fieldset><div></div><div></div></div><div class=\\"button-block\\"><div class=\\"confirmator\\"><button type=\\"submit\\" class=\\"confirmator-button\\">&#x414;&#x43E;&#x431;&#x430;&#x432;&#x438;&#x442;&#x44C;</button></div></div></form></div>"`;

exports[`snapshot testing LinkAdder рендерится правильно 1`] = `"<div class=\\"component-body\\"></div>"`;

exports[`snapshot testing LinkList рендерится правильно 1`] = `"<div class=\\"component-header\\"><h2 class=\\"component-title\\">&#x421;&#x441;&#x44B;&#x43B;&#x43A;&#x438;</h2><input type=\\"search\\" class=\\"link-filter\\" placeholder=\\"&#x424;&#x438;&#x43B;&#x44C;&#x442;&#x440;\\"></div><div class=\\"component-body\\"><div class=\\"component-body__top-section\\"><div class=\\"button-block\\"><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><span><i class=\\"fas fa-plus-circle\\"></i>&#x414;&#x43E;&#x431;&#x430;&#x432;&#x438;&#x442;&#x44C;</span></button><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><i class=\\"fas fa-copy\\"></i>&#x421;&#x43A;&#x43E;&#x43F;&#x438;&#x440;&#x43E;&#x432;&#x430;&#x442;&#x44C; &#x432; &#x431;&#x443;&#x444;&#x435;&#x440; <span class=\\"help-text\\">&lt;Ctrl&gt;+&lt;C&gt;</span></button><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><i class=\\"fas fa-align-justify\\"></i>&#x421;&#x43A;&#x430;&#x447;&#x430;&#x442;&#x44C; &#x441;&#x441;&#x44B;&#x43B;&#x43A;&#x438; <span class=\\"help-text\\">&lt;Ctrl&gt;+&lt;S&gt;</span></button></div><div class=\\"link-list__tabs button-block\\"><button class=\\" active\\" data-target=\\"&#x412;&#x441;&#x435;\\">&#x412;&#x441;&#x435;</button><button class></button></div><div class=\\"resource plaque warning\\"><i class=\\"fas fa-times-circle\\"></i>&#x41D;&#x435;&#x442; &#x441;&#x441;&#x44B;&#x43B;&#x43E;&#x43A;</div></div></div>"`;

exports[`snapshot testing LinkManager рендерится правильно 1`] = `"<div id=\\"notification-overlay\\"></div><div class=\\"flex-wrapper\\"><div class=\\"link-list \\"><div class=\\"component-header\\"><h2 class=\\"component-title\\">&#x421;&#x441;&#x44B;&#x43B;&#x43A;&#x438;</h2><input type=\\"search\\" class=\\"link-filter\\" placeholder=\\"&#x424;&#x438;&#x43B;&#x44C;&#x442;&#x440;\\"></div><div class=\\"component-body\\"><div class=\\"component-body__top-section\\"><div class=\\"button-block\\"><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><span><i class=\\"fas fa-plus-circle\\"></i>&#x414;&#x43E;&#x431;&#x430;&#x432;&#x438;&#x442;&#x44C;</span></button><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><i class=\\"fas fa-copy\\"></i>&#x421;&#x43A;&#x43E;&#x43F;&#x438;&#x440;&#x43E;&#x432;&#x430;&#x442;&#x44C; &#x432; &#x431;&#x443;&#x444;&#x435;&#x440; <span class=\\"help-text\\">&lt;Ctrl&gt;+&lt;C&gt;</span></button><button style=\\"float:none\\" class=\\" waves-effect hoverable waves-light btn add-button\\" type=\\"button\\"><i class=\\"fas fa-align-justify\\"></i>&#x421;&#x43A;&#x430;&#x447;&#x430;&#x442;&#x44C; &#x441;&#x441;&#x44B;&#x43B;&#x43A;&#x438; <span class=\\"help-text\\">&lt;Ctrl&gt;+&lt;S&gt;</span></button></div><div class=\\"link-list__tabs button-block\\"><button class=\\" active\\" data-target=\\"&#x412;&#x441;&#x435;\\">&#x412;&#x441;&#x435;</button></div><div class=\\"resource plaque warning\\"><i class=\\"fas fa-times-circle\\"></i>&#x41D;&#x435;&#x442; &#x441;&#x441;&#x44B;&#x43B;&#x43E;&#x43A;</div></div></div></div><div class=\\"link-adder\\"><div class=\\"component-body\\"></div></div></div>"`;

exports[`snapshot testing LinkResource рендерится правильно 1`] = `"<div class=\\"component-header component-header--subcomponent\\"><h2 class=\\"component-title\\">&#x412;&#x44B;&#x431;&#x440;&#x430;&#x43D;&#x43D;&#x44B;&#x435; &#x440;&#x435;&#x441;&#x443;&#x440;&#x441;&#x44B;</h2></div><div class=\\"component-body component-body--subcomponent\\"><div class=\\"resource plaque warning\\"><i class=\\"fas fa-times-circle left-icon\\"></i>&#x41D;&#x435; &#x432;&#x44B;&#x431;&#x440;&#x430;&#x43D; &#x440;&#x435;&#x441;&#x443;&#x440;&#x441;</div></div>"`;

exports[`snapshot testing ResourceExplorer рендерится правильно 1`] = `"<div class=\\"component-header component-header--subcomponent\\"><h2 class=\\"component-title\\">&#x41F;&#x43E;&#x438;&#x441;&#x43A; &#x440;&#x435;&#x441;&#x443;&#x440;&#x441;&#x430;</h2></div><div class=\\"resource-search-form\\"><div class=\\"form-wrapper\\"><form class=\\"rjsf\\"><div class=\\"form-group field field-object\\"><fieldset id=\\"root\\"><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_name\\">&#x41D;&#x430;&#x437;&#x432;&#x430;&#x43D;&#x438;&#x435; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_name\\" label=\\"&#x41D;&#x430;&#x437;&#x432;&#x430;&#x43D;&#x438;&#x435; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_parent_name\\">&#x41F;&#x430;&#x43F;&#x43A;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_parent_name\\" label=\\"&#x41F;&#x430;&#x43F;&#x43A;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_item_search_search_nested\\" value=\\"true\\"></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_code\\">&#x41A;&#x43E;&#x434; 1&#x421; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_code\\" label=\\"&#x41A;&#x43E;&#x434; 1&#x421; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_article\\">&#x410;&#x440;&#x442;&#x438;&#x43A;&#x443;&#x43B;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_article\\" label=\\"&#x410;&#x440;&#x442;&#x438;&#x43A;&#x443;&#x43B;\\" placeholder><div></div><div></div></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_id\\" value></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_preset\\" value=\\"0\\"></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_type\\" value=\\"1\\"></div></fieldset><div></div><div></div></div><div class=\\"button-block\\"><button type=\\"submit\\"><span><i class=\\"fas fa-search\\"></i>&#x41F;&#x43E;&#x438;&#x441;&#x43A;</span></button></div></form></div></div><div class=\\"resource-search-results\\"><div class=\\"component-header component-header--subcomponent\\"><h2 class=\\"component-title\\">&#x420;&#x435;&#x437;&#x443;&#x43B;&#x44C;&#x442;&#x430;&#x442;&#x44B; &#x43F;&#x43E;&#x438;&#x441;&#x43A;&#x430;</h2></div><div class=\\"component-body component-body--subcomponent\\"><div class=\\"search-results\\">&#x41D;&#x435;&#x442; &#x440;&#x435;&#x441;&#x443;&#x440;&#x441;&#x43E;&#x432;</div></div></div>"`;

exports[`snapshot testing ResourceSearchForm рендерится правильно 1`] = `"<div class=\\"form-wrapper\\"><form class=\\"rjsf\\"><div class=\\"form-group field field-object\\"><fieldset id=\\"root\\"><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_name\\">&#x41D;&#x430;&#x437;&#x432;&#x430;&#x43D;&#x438;&#x435; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_name\\" label=\\"&#x41D;&#x430;&#x437;&#x432;&#x430;&#x43D;&#x438;&#x435; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_parent_name\\">&#x41F;&#x430;&#x43F;&#x43A;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_parent_name\\" label=\\"&#x41F;&#x430;&#x43F;&#x43A;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_item_search_search_nested\\" value=\\"true\\"></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_code\\">&#x41A;&#x43E;&#x434; 1&#x421; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_code\\" label=\\"&#x41A;&#x43E;&#x434; 1&#x421; &#x442;&#x43E;&#x432;&#x430;&#x440;&#x430;\\" placeholder><div></div><div></div></div><div class=\\"form-group field field-string\\"><label class=\\"control-label\\" for=\\"root_item_search_article\\">&#x410;&#x440;&#x442;&#x438;&#x43A;&#x443;&#x43B;</label><div></div><input type=\\"text\\" class=\\"form-control\\" value id=\\"root_item_search_article\\" label=\\"&#x410;&#x440;&#x442;&#x438;&#x43A;&#x443;&#x43B;\\" placeholder><div></div><div></div></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_id\\" value></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_preset\\" value=\\"0\\"></div><div class=\\"hidden\\"><input type=\\"hidden\\" id=\\"root_resource_search_type\\" value=\\"1\\"></div></fieldset><div></div><div></div></div><div class=\\"button-block\\"><button type=\\"submit\\"><span><i class=\\"fas fa-search\\"></i>&#x41F;&#x43E;&#x438;&#x441;&#x43A;</span></button></div></form></div>"`;

exports[`snapshot testing ResourceSearchResults рендерится правильно 1`] = `"<div class=\\"component-header component-header--subcomponent\\"><h2 class=\\"component-title\\">&#x420;&#x435;&#x437;&#x443;&#x43B;&#x44C;&#x442;&#x430;&#x442;&#x44B; &#x43F;&#x43E;&#x438;&#x441;&#x43A;&#x430;</h2></div><div class=\\"component-body component-body--subcomponent\\"><div class=\\"search-results\\">&#x41D;&#x435;&#x442; &#x440;&#x435;&#x441;&#x443;&#x440;&#x441;&#x43E;&#x432;</div></div>"`;
